src/pydot/core.py

```diff
@@ -569,6 +569,22 @@ def set_parent_graph(self, parent_graph):
     def get_parent_graph(self):
         return self.obj_dict.get("parent_graph", None)
 
+    def get_top_graph_type(self):
+        """Find the topmost parent graph type for the current object."""
+        if isinstance(self, Graph):
+            parent = self
+        else:
+            parent = self.get_parent_graph()
+        if parent is None:
+            return None
```

Comment:
Now that I think about it, I can simplify this. We know that `Graph.get_parent_graph()` will always return the graph itself, so it'll just short-circuit the loop sooner.

```suggestion
    def get_top_graph_type(self):
        """Find the topmost parent graph type for the current object."""
        parent = self.get_parent_graph()
        if parent is None:
            return None
```